### PR TITLE
Adding RestActions support for Detector Stats API

### PR DESCRIPTION
**Pull request by saratvemulapalli**
*Issue #195 *

*Description of changes:*
Adding RestActions support Detector Stats API which is needed for FGAC support. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by saratvemulapalli** [comment]
Adding Unit Tests in the coming commit.

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/SearchAnomalyDetectorAction.java [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
 public class SearchAnomalyDetectorAction extends ActionType<SearchResponse> {
     public static final SearchAnomalyDetectorAction INSTANCE = new SearchAnomalyDetectorAction();
-    public static final String NAME = "cluster:admin/opendistro/ad/detector/search";
+    public static final String NAME = "cluster:admin/ad/search/detector";
```
Comment (kaituo) [inline src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/SearchAnomalyDetectorAction.java]: The old name sound more reasonable.

Comment (saratvemulapalli) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/SearchAnomalyDetectorAction.java]: Sure, that was some merge issue which showed up here. 
Cleaned it up with another commit.

File: build.gradle [hunk 2]
```diff
@@ -249,6 +249,7 @@ List<String> jacocoExclusions = [
         'com.amazon.opendistroforelasticsearch.ad.AnomalyDetectorRunner',
         'com.amazon.opendistroforelasticsearch.ad.indices.AnomalyDetectionIndices',
         'com.amazon.opendistroforelasticsearch.ad.util.ParseUtils',
+        'com.amazon.opendistroforelasticsearch.ad.transport.StatsAnomalyDetectorTransportAction'
```
Comment (saratvemulapalli) [inline build.gradle:252]: As we are already covering them in our IntegTests, its getting complicated to unit test it.

**Review by skkosuri-amzn — APPROVED** [review]

**Review by kaituo — APPROVED** [review]

**Pull request merged**